Sources/NIO/MarkedCircularBuffer.swift

```diff
@@ -161,6 +161,13 @@ extension MarkedCircularBuffer: Collection, MutableCollection {
         get {
             return self._buffer[bounds]
         }
+        set {
+            var index = bounds.lowerBound
+            for newElement in newValue where index < bounds.upperBound {
```

Comment:
As you're using `formIndex(after:)` this can just be a `while` loop, which terminates when `index == bounds.upperBound`.

Reply:
What I wanted to do was use `zip`, but the type of `MarkedCircularBuffer.Index` wouldn't allow that. The loop needs two iterators either way, so using it seems to me that using a `while` would just make both iterations explicit. Thoughts?

Reply:
After peeking, it looks like what you're saying is that `==` would be faster than `<`.